modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/ImportWORMBASE.pm

```diff
@@ -0,0 +1,183 @@
+ <http://www.ensembl.org/Help/Contact>.
+
+=cut
+
+
+=head1 ImportWORMBASE
+
+This module imports WORMBASE Celegans phenotype data.
+
+=cut
+
+package Bio::EnsEMBL::Variation::Pipeline::PhenotypeAnnotation::ImportWORMBASE;
+
+use strict;
+use warnings;
+
+use File::Path qw(make_path);
+use File::stat;
+use POSIX qw(strftime);
+use LWP::Simple;
+
+use base ('Bio::EnsEMBL::Variation::Pipeline::PhenotypeAnnotation::BasePhenotypeAnnotation');
+
+my %source_info;
+
+sub fetch_input {
+  #create output folder structure and fetches input files
+  my $self = shift;
+
+  my $pipeline_dir    = $self->required_param('pipeline_dir');
+  my $species         = $self->required_param('species');
+
+  $self->debug($self->param('debug_mode'));
+  $self->core_db_adaptor($self->get_species_adaptor('core'));
+  $self->variation_db_adaptor($self->get_species_adaptor('variation'));
+
+  # import specific constants
+  %source_info = (source_description => 'WormBase Phenotypes',
+                  source_url => 'https://wormbase.org/species/all/phenotype/',
+                  object_type => 'Gene',
+                  source_version => "$ENV{WORMBASE_VERSION}",
+
+                  source_status => 'somatic',
+
+                  source_name => 'WORMBASE_PHENOTYPE',        #source name in the variation db
+                  source_name_short => 'WormBase',  #source identifier in the pipeline
+                  data_types => 'phenotype_feature',
+                  );
+
+  #create workdir folder
+  my $workdir = $pipeline_dir."/".$source_info{source_name}."/".$species;
+  unless (-d $workdir) {
+    make_path($workdir) or die "Failed to create $workdir $!\n";
+  }
+  $self->workdir($workdir);
+
+  open(my $logFH, ">", $workdir."/".'log_import_out_'.$source_info{source_name}.'_'.$species) || die ("Failed to open file: $!\n");
+  open(my $errFH, ">", $workdir."/".'log_import_err_'.$source_info{source_name}.'_'.$species) || die ("Failed to open file: $!\n");
+  open(my $pipelogFH, ">", $workdir."/".'log_import_debug_pipe_'.$source_info{source_name}.'_'.$species) || die ("Failed to open file: $!\n");
+  $self->logFH($logFH);
+  $self->errFH($errFH);
+  $self->pipelogFH($pipelogFH);
+
+  #find data file
+  my $inputFile = 'wb_phenotypes.txt';
+  die "expected $workdir/$inputFile" unless -e $workdir."/".$inputFile;
+
+  $self->param('RNAi_file', $inputFile);
+
+}
+#
+sub run {
+  my $self = shift;
+  my $input_file = $self->required_param('RNAi_file');
+
+  # get phenotype data
+  my $results = $self->parse_input_file($input_file);
+  $self->print_logFH("Got ".(scalar @{$results->{'phenotypes'}})." phenotypes \n") if ($self->debug);
+
+  # save phenotypes
+  $self->save_phenotypes(\%source_info, $results);
+
+  my %param_source = (source_name => $source_info{source_name},
+                      type => $source_info{object_type});
+
+  $self->param('output_ids', { source => \%param_source,
+                               species => $self->required_param('species')
```

Comment:
run_type is necessary for the next module check_phenotypes.
`run_type => $self->required_param('run_type')`

Reply:
Should i just add this line within the run subroutine?

Reply:
Yes, you can add it after `species => $self->required_param('species')`.

Reply:
Should i just add this line under the run subroutine?

Reply:
It's inside the subroutine run:  
```
$self->param('output_ids', { source => \%param_source,
                               species => $self->required_param('species'),
                               run_type => $self->required_param('run_type')
  });
```

Reply:
Done.